Update installation instructions (#1039)
haddocking/haddock3: README.md

Reviewer: I would add the disclaimer also here (or a link to it), to make understand that by running those commands, they are already bound to the CNS license.

## Installation

> Clone the repository and run `pip install`;

Reviewer: ```suggestion
Clone the repository and run `pip install .`;
```

```bash
git clone https://github.com/haddocking/haddock3
cd haddock3
pip install .
```

Execute:

```bash
$ haddock3 -h

> usage: haddock3 [-h] [--restart RESTART] [--extend-run EXTEND_RUN] [--setup] [--log-level {DEBUG,INFO,WARNING,ERROR,CRITICAL}] [-v] recipe
> 
> positional arguments:
>   recipe                The input recipe file path

Reviewer: ```suggestion
usage: haddock3 [-h] [--restart RESTART] [--extend-run EXTEND_RUN] [--setup] [--log-level {DEBUG,INFO,WARNING,ERROR,CRITICAL}] [-v] workflow

positional arguments:
  workflow                The input configuration file path describing the workflow to be performed
```

Author: This is coming from the CLI descriptions - will keep it "wrong" just to be consistent

Reviewer: Lets modify the CLI then ?
`recipe` is to ambiguous to understand, we are not doing cooking here

Reviewer: Modified in #1043

optional arguments:
  -h, --help            show this help message and exit

>   --restart RESTART     Restart the run from a given step. Previous folders from the selected step onward will be deleted.

Reviewer 2: ```suggestion
  --restart RESTART     Restart the run from a given step. Previous folders from the selected step onwards will be deleted.
```

Author: this is a copy from what's printed with the `-h` argument, maybe good to also fix it there but just a minor typo
